src/eva/data/data_collections.py

```diff
@@ -66,45 +76,116 @@ def create_or_add_to_collection(self, collection_name, collection, concat_dimens
+
+        variable_data = variable_array.data
+
+        # Squeeze in case of dimension of 1 (e.g. when 1 channel is needed)
+        variable_data = np.squeeze(variable_data)
+
+        return variable_data
+
+    # ----------------------------------------------------------------------------------------------
+
+    def validate_names(self):
+
+        # This code checks that the naming conventions are compliant with what is expected
+
+        for collection_key in self._collections.keys():
 
-        return self._collections[collection][variable].data
+            # Assert that the collection name does not contain disallowed characters
+            if not string_does_not_contain(disallowed_chars, collection_key):
+                self.logger.abort(f'Collection contains the key \'{collection_key}\', which ' +
+                                  f'contains a characeter that is not permitted ' +
+                                  f'({disallowed_chars})')
+
+            # Loop over the data variables
+            for data_var in list(self._collections[collection_key].data_vars):
+
+                # Assert that the datavar contains '::' identifier, splitting group and variable
+                if '::' not in data_var:
+                    self.logger.abort(f'Collection \'{collection_key}\' contains the following ' +
+                                      f'data variable \'{data_var}\', which does not contain ' +
+                                      f'\'::\' splitting the group and variable.')
+                [group, variable] = data_var.split('::')
+                # Assert that the group name does not contain disallowed characters
+                if not string_does_not_contain(disallowed_chars, group):
+                    self.logger.abort(f'Collection \'{collection_key}\' contains the following ' +
+                                      f'element \'{data_var}\'. The group \'{group}\'' +
+                                      f'contains a character that is not permitted ' +
+                                      f'({disallowed_chars}).')
+                # Assert that the variable name does not contain disallowed characters
+                if not string_does_not_contain(disallowed_chars, variable):
+                    self.logger.abort(f'Collection \'{collection_key}\' contains the following ' +
+                                      f'element \'{data_var}\'. The variable \'{variable}\'' +
+                                      f'contains a character that is not permitted ' +
+                                      f'({disallowed_chars}).')
 
     # ----------------------------------------------------------------------------------------------
 
-    def __str__(self):
+    def display_collections(self):
 
-        # Print a list of variables that are available in the collection
-        self.logger.info("Collections available: ", )
+        # Display a list of variables that are available in the collection
+        self.logger.info('-'*80)
```

Comment:
Does this print 80 `-`? I've never known about this syntax before!

Reply:
It does. The wonderful world of Python.